snapcraft/internal/sources/_bazaar.py

```diff
@@ -37,6 +37,9 @@ def __init__(self, source, source_dir, source_tag=None, source_commit=None,
             raise errors.IncompatibleOptionsError(
                 'can\'t specify both source-tag and source-commit for '
                 'a bzr source')
+        if source_checksum:
+            raise errors.IncompatibleOptionsError(
+                'can\'t specify a source-checksum for a bzr source')
```

Comment:
tiny detail, it's more readable like:
"can't specify a source ..."
It's the only exception where we like double quotes. I know the lines above are not following it, we can fix them later.